File: python/euler/bin/p0025.py
# project euler: problem 25


def compute(digits: int) -> str:
    limit = pow(10, digits - 1)
    nth = 2
    fib1, fib2 = 1, 1
    while fib2 < limit:
        nth += 1
        fib1, fib2 = fib2, fib2 + fib1

    return str(nth)


def solve() -> str:
    return compute(1_000)

File: typescript/euler/bin/p0025.ts
// project euler: problem 25

export const compute = (digit: number): string => {
  const limit = 10n ** (BigInt(digit) - 1n);
  let nth = 2;
  let fib1 = 1n, fib2 = 1n;
  while (fib2 < limit) {
    nth += 1;
    [fib1, fib2] = [fib2, fib1 + fib2];
  }

  return String(nth);
};

export const solve = (): string => compute(1_000);
